CMakeLists.txt

```diff
@@ -113,6 +113,12 @@ if (NOT MSVC)
   set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -DDEBUG -O0 -g3 -ggdb")
 endif(NOT MSVC)
 
+# Use Address Sanitizer(ASAN) to check memory error, only happens when -DCMAKE_BUILD_TYPE=DEBUG
+if((NOT MSVC) AND USE_ASAN)
+  message(STATUS "Use ASAN")
+  set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -fsanitize=address -fno-omit-frame-pointer -fsanitize-recover=all -fsanitize=leak")
```

Comment:
please enable here: https://github.com/dmlc/dgl/blob/7467da9a39dd194eec4e0df24dc9689dd60a0ac4/CMakeLists.txt#L118.

Reply:
done

Reply:
There is still a [memory leak](https://github.com/dmlc/dgl/pull/6141/files#diff-af1b783953c1d9d9eb76f303b7b5227a61b6b1ee6b4c074146488cc2de9d6cc9R131) in the cpp unit test , so, if I enable here, the UT will be failed. Now, I put the option "USE_ASAN" in a separate scope, when the UT fixed, I'll enable it here.